### ISSUE TITLE
Need to Add Dark mode Functionality to app

### ISSUE THREAD

**Issue by adarsh-technocrat** [codeforcauseorg/edu-client#147]
## Is your feature request related to a problem? Please describe.

As profile page UI is been implemented to the application now we can go for implementing functionality to it. We can go for enabling the Dark Mode Feature which is the most common use case of the application which can be controlled from the profile page.

## Describe the solution you'd like
The Approach to solve this issue would be:
1. Setting up the [theme config ](https://github.com/codeforcauseorg/edu-client/blob/a7304077d8d26ef199277335702e8cbeec87d6a4/src/theme/index.js#L29)   to support Dark theme currently palette only support the light theme.

2. Creating `themeReducers.js` file in  `src/reducers/`  so that we can use reducer value to switch between Darke and Light Theme using [Switch](https://github.com/codeforcauseorg/edu-client/blob/a7304077d8d26ef199277335702e8cbeec87d6a4/src/pages/ProfilePage/ProfileComponent/ListComponent.js#L122)

3.  At last we need to save the user Preference state for that we can use local storage so when every user comes back to the application they can see their preferred state.

 **This could be one of the ways to solve the issue if anyone has better would love to hear that**  

## Additional context

Add any other context about the feature request here.

**Comment by kumard3** [comment]
Hey @adarsh-technocrat , I would like to work on this.  Just assign me the problem so that I can work accordingly.

**Comment by kumard3** [comment]
Hey @adarsh-technocrat , I would like to work on this.

**Comment by adarsh-technocrat** [comment]
> Hey @adarsh-technocrat , I would like to work on this.

Thanks @kumard3  for your interest to the issue but I was waiting for this issue could be asign to me :)

**Comment by Siddhant-K-code** [comment]
yes, I think @adarsh-technocrat made his implementation very clear, so it could get assigned to him. 

/cc @Abhishek-kumar09 @KeenWarrior
